src/test/java/seedu/address/logic/commands/MarkCommandTest.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code MarkCommand}.
+ */
+public class MarkCommandTest {
+
+    private Model model = new ModelManager(getTypicalTaskList(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Task taskToMark = model.getFilteredTaskList().get(INDEX_FIRST_TASK.getZeroBased());
+        Task markedTask = MarkCommand.createMarkedTask(taskToMark);
+        MarkCommand markCommand = new MarkCommand(INDEX_FIRST_TASK);
+
+        String expectedMessage = String.format(MarkCommand.MESSAGE_MARK_TASK_SUCCESS, markedTask);
+
+        ModelManager expectedModel = new ModelManager(model.getTaskList(), new UserPrefs());
+        expectedModel.strictSetTask(model.getFilteredTaskList().get(0), markedTask);
+
+        assertCommandSuccess(markCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredTaskList().size() + 1);
+        MarkCommand markCommand = new MarkCommand(outOfBoundIndex);
+
+        assertCommandFailure(markCommand, model, Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validCompletedTaskUnfilteredList_throwsCommandException() {
+        // INDEX SECOND TASK: default completion status set to true
+        Task taskToMark = model.getFilteredTaskList().get(INDEX_SECOND_TASK.getZeroBased());
+        assertEquals(taskToMark.getCompletionStatus().value, "true");
+
+        MarkCommand markCommand = new MarkCommand(INDEX_SECOND_TASK);
+
+        assertCommandFailure(markCommand, model, MarkCommand.MESSAGE_TASK_ALREADY_COMPLETED);
+    }
+
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        showTaskAtIndex(model, INDEX_FIRST_TASK);
+
+        Task taskToMark = model.getFilteredTaskList().get(INDEX_FIRST_TASK.getZeroBased());
+        Task markedTask = MarkCommand.createMarkedTask(taskToMark);
+        MarkCommand markCommand = new MarkCommand(INDEX_FIRST_TASK);
+
+        String expectedMessage = String.format(MarkCommand.MESSAGE_MARK_TASK_SUCCESS, markedTask);
+
+        ModelManager expectedModel = new ModelManager(model.getTaskList(), new UserPrefs());
+        expectedModel.strictSetTask(model.getFilteredTaskList().get(0), markedTask);
+
+        assertCommandSuccess(markCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showTaskAtIndex(model, INDEX_FIRST_TASK);
+
+        Index outOfBoundIndex = INDEX_SECOND_TASK;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getTaskList().getTaskList().size());
+
+        MarkCommand markCommand = new MarkCommand(outOfBoundIndex);
```

Comment:
I'm not entirely sure about this but should the filtered task list be called within this function since the function? (Because the markcommand should throw an error when marking an outOfBoundIndex wrt the filtered list)